src/angular/processflow/processflow.ts

```diff
@@ -0,0 +1,171 @@
+    @SkipSelf() private _errorStateMatcher: SbbErrorStateMatcher,
+    private _viewContainerRef: ViewContainerRef,
+    @Optional() @Inject(STEPPER_GLOBAL_OPTIONS) processflowOptions?: StepperOptions
+  ) {
+    super(processflow, processflowOptions);
+  }
+
+  ngAfterContentInit() {
+    this._isSelected = this._stepper.steps.changes
+      .pipe(
+        switchMap(() => {
+          return this._stepper.selectionChange.pipe(
+            map((event) => event.selectedStep === this),
+            startWith(this._stepper.selected === this)
+          );
+        })
+      )
+      .subscribe((isSelected) => {
+        if (isSelected && this._lazyContent && !this._portal) {
+          this._portal = new TemplatePortal(this._lazyContent._template, this._viewContainerRef!);
+        }
+      });
+  }
+
+  ngOnDestroy() {
+    this._isSelected.unsubscribe();
+  }
+
+  /** Custom error state matcher that additionally checks for validity of interacted form. */
+  isErrorState(control: FormControl | null, form: FormGroupDirective | NgForm | null): boolean {
+    const originalErrorState = this._errorStateMatcher.isErrorState(control, form);
+
+    // Custom error state checks for the validity of form that is not submitted or touched
+    // since user can trigger a form change by calling for another step without directly
+    // interacting with the current form.
+    const customErrorState = !!(control && control.invalid && this.interacted);
+
+    return originalErrorState || customErrorState;
+  }
+}
+
+@Component({
+  selector: 'sbb-processflow, [sbbProcessflow]',
+  exportAs: 'sbbProcessflow',
+  templateUrl: 'processflow.html',
+  styleUrls: ['processflow.css'],
+  inputs: ['selectedIndex'],
+  host: {
+    class: 'sbb-processflow',
+    '[attr.aria-orientation]': 'orientation',
+    role: 'tablist',
+  },
+  animations: [sbbProcessflowAnimations.stepTransition],
+  providers: [{ provide: CdkStepper, useExisting: SbbProcessflow }],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SbbProcessflow extends CdkStepper implements AfterContentInit {
+  /** The list of step headers of the steps in the processflow. */
+  @ViewChildren(SbbStepHeader) _stepHeader: QueryList<SbbStepHeader>;
+
+  /** Full list of steps inside the processflow, including inside nested processflows. */
+  @ContentChildren(SbbStep, { descendants: true }) _steps: QueryList<SbbStep>;
+
+  /** Steps that belong to the current processflow, excluding ones from nested processflows. */
+  readonly steps: QueryList<SbbStep> = new QueryList<SbbStep>();
+
+  /** Event emitted when the current step is done transitioning in. */
+  @Output() readonly animationDone: EventEmitter<void> = new EventEmitter<void>();
+
+  /** Stream of animation `done` events when the body expands/collapses. */
+  readonly _animationDone = new Subject<AnimationEvent>();
+
+  constructor(
+    @Optional() dir: Directionality,
```

Comment:
Due to the CDKStepper it's necessary to inject dir? Or can we just pass null to super (which I think would be ok, because its optional)